core/src/main/java/org/apache/gravitino/stats/storage/LancePartitionStatisticStorage.java

```diff
@@ -197,123 +255,131 @@ public void updateStatistics(
+  @Override
+  public void close() throws IOException {
+    if (allocator != null) {
+      allocator.close();
     }
+
+    if (cache != null) {
+      cache.invalidateAll();
+    }
+  }
+
+  @VisibleForTesting
+  Cache<Long, Dataset> getCache() {
+    return cache;
   }
 
   private String getFilePath(Long tableId) {
     return location + "/" + tableId + ".lance";
   }
 
-  private void dropStatisticsImpl(Long tableId, List<PartitionStatisticsDrop> drops) {
-    String fileName = getFilePath(tableId);
-    try (Dataset dataset = open(fileName)) {
-      List<String> partitionSQLs = Lists.newArrayList();
-      for (PartitionStatisticsDrop drop : drops) {
-        List<String> statistics = drop.statisticNames();
-        String partition = drop.partitionName();
-        partitionSQLs.add(
-            "table_id = "
-                + tableId
-                + " AND partition_name = '"
-                + partition
-                + "' AND statistic_name IN ("
-                + statistics.stream().map(str -> "'" + str + "'").collect(Collectors.joining(", "))
-                + ")");
+  private List<FragmentMetadata> createFragmentMetadata(
+      Long tableId, List<PartitionStatisticsUpdate> updates) throws JsonProcessingException {
+    List<FragmentMetadata> fragmentMetas;
+    int count = 0;
+    try (VectorSchemaRoot root = VectorSchemaRoot.create(SCHEMA, allocator)) {
+      for (PartitionStatisticsUpdate update : updates) {
+        count += update.statistics().size();
       }
 
-      if (partitionSQLs.size() == 1) {
-        dataset.delete(partitionSQLs.get(0));
-      } else if (partitionSQLs.size() > 1) {
-        String filterSQL =
-            partitionSQLs.stream().map(str -> "(" + str + ")").collect(Collectors.joining(" OR "));
-        dataset.delete(filterSQL);
+      for (FieldVector vector : root.getFieldVectors()) {
+        vector.setInitialCapacity(count);
+      }
+      root.allocateNew();
+      int index = 0;
+
+      for (PartitionStatisticsUpdate updatePartitionStatistic : updates) {
+        String partitionName = updatePartitionStatistic.partitionName();
+        for (Map.Entry<String, StatisticValue<?>> statistic :
+            updatePartitionStatistic.statistics().entrySet()) {
+          String statisticName = statistic.getKey();
+          String statisticValue =
+              JsonUtils.anyFieldMapper().writeValueAsString(statistic.getValue());
+
+          UInt8Vector tableIdVector = (UInt8Vector) root.getVector(TABLE_ID_COLUMN);
+          VarCharVector partitionNameVector = (VarCharVector) root.getVector(PARTITION_NAME_COLUMN);
+          VarCharVector statisticNameVector = (VarCharVector) root.getVector(STATISTIC_NAME_COLUMN);
+          LargeVarCharVector statisticValueVector =
+              (LargeVarCharVector) root.getVector(STATISTIC_VALUE_COLUMN);
+          VarCharVector auditInfoVector = (VarCharVector) root.getVector(AUDIT_INFO_COLUMN);
```

Comment:
The lines can be moved out of the for loop according to the code below. Is it expected?